### PR TITLE
AudioTrack: Simplify Channel selection

### LINKED ISSUE #24546: Kodi Omega Android Nightly - Quad plays as 'Dolby Audio' (stereo only)
### Describe the bug
Here is a clear and concise description of what the problem is:
I have recently started testing various surround/spatial audio formats on a new Amazon Firestick 4K Max (2nd gen 2023) using latest Kodi v21 Omega Android nightlies.

When I play  Quad FLAC music file with less than 6 channels my AVR shows it is receiving Stereo only (the Denon AVR shows 'Dolby Audio') which then is upmixed by the AVR - wrong).

This is a critical bug for users with Quad music collections.

## Expected Behavior
Here is a clear and concise description of what was expected to happen:
All Quad formatted files should play as multi-channel files (not stereo) with the discreet channels sent to the AVR, just like they always have in previous Kodi releases (and do now with LibreElec v21 nightly)

## Actual Behavior
Quad in a 5.1 formatted file (Quad + Silent C and Silent LFE) - this plays a quad on the FTV (correct)
Quad in a 5.0 formatted file (Quad + Silent C) - stereo output only (wrong)
Quad in a 4.0 formatted file (Quad with channel mask L, R, Ls, Rs) - stereo output only (wrong)
Quad in a 4.0 formatted file (Quad with channel mask L, R, Lb Rb - stereo output only (wrong)

## Possible Fix
These exact same files ALL play correctly in LibreElec v12 Beta 2 Generic (Kodi v21 Omega build 17 Jan). These same files have always played correctly in recent Kodi releases (Matrix etc). I assume the Kodi IEC Packer needs a fix (unsure).

### To Reproduce
Steps to reproduce the behaviour:
I have created 4 x 30 sec FLAC test files with the typical channel layouts used by Quad enthusiasts over the years:

Link to zip file containing 4 x FLACs and a FTV Kodi.log file: (78MB)
https://1drv.ms/u/s!Aqs3ia6BNDDHsAwowhnjZHCusXcy?e=zTJXy7

Play these on a Fire TV 4K Max with Kodi Omega Nightly (I suspect other Android devices may have same issue)

### Debuglog
The debuglog can be found here:
A Kodi.log file is in the test file zip linked above:

### Screenshots 
Here are some links or screenshots to help explain the problem:
(none)

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
Historically  Quad enthusiasts have decoded 4 channel FLAC. Mnay AVR could not play pur 4 channel files so the 4 different formats in the test files are all very commonly used. Kodi needs to support all 4 to be used in the quad community. (it always has BTW)

### Your Environment
Used Operating system:

 - [X ] Android  (Amazon Fire TV 4K Max @nd Gen (2023 model)
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name:
 - Kodi version:
kodi-20240118-65630714-master-armeabi-v7a.apk

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

### PR DESCRIPTION
**Pull request by fritsch**
All major Android devices have issues opening more "fine grain" audio channel configurations besides Stereo, 5.1 or .7.1 - use AE's feature to silently add additional channels so that e.g. 3.0, 4.0, 4.1, etc. are tunneled via 5.1, 7.1 and rest is sent via stereo.

The issue is here not kodi, but Android not being able to open the mask, throwing channels away while accepting the input without error.

Fixes:  #24546

### CODE REVIEW

File: xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp [hunk 1]
```diff
@@ -130,26 +130,18 @@ static int AEChannelMapToAUDIOTRACKChannelMask(CAEChannelInfo info)
...
-    return CJNIAudioFormat::CHANNEL_OUT_7POINT1_SURROUND;
+  if (info.Count() > 6)
+   return CJNIAudioFormat::CHANNEL_OUT_7POINT1_SURROUND;
 
-  int atMask = 0;
+  if (info.Count() > 2)
+    return CJNIAudioFormat::CHANNEL_OUT_5POINT1;
 
-  for (unsigned int i = 0; i < info.Count(); i++)
-    atMask |= AEChannelToAUDIOTRACKChannel(info[i]);
```
Comment (thexai) [inline xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:150]: Jenkins Android builds fails due new warning generated:

```
/home/jenkins/jenkins-root/workspace/android-arm-docker@2/xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:93:12: warning: unused function 'AEChannelToAUDIOTRACKChannel' [-Wunused-function]
static int AEChannelToAUDIOTRACKChannel(AEChannel aeChannel)
           ^
1 warning generated.
```

Comment (fritsch) [reply xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:150]: I can remove this function as well. Will make a second commit though, as we need it back when we want to do it more properly again in the future.

**Review by thexai — APPROVED** [review]
Tested on Shield and confirmed that now outputs muti-channel PCM with FLAC 4.0 and 4.1 sample files.

**Comment by fritsch** [comment]
So - whatever comes up as regressions I will take up directly, which is why I did not wait longer with merging.

**Pull request merged**